src/ln/channel.rs

```diff
@@ -1300,6 +1300,9 @@ impl Channel {
 		if self.channel_outbound {
 			return Err(HandleError{err: "Received funding_created for an outbound channel?", action: Some(msgs::ErrorAction::SendErrorMessage {msg: msgs::ErrorMessage {channel_id: self.channel_id, data: "Received funding_created for an outbound channel?".to_string()}})});
 		}
+		if self.channel_id != msg.temporary_channel_id {
```

Comment:
The method is called on self b/c this condition is true. So this if expression will never be false.

Reply:
I potentially have this wrong, but isnt msg the incoming message from the remote peer? Which means h=it could have it wrong

Reply:
channelmanager maintains a mapping from channel_id to channel struct, so if an unknown temporary_channel_id is received, it will be disregarded instead of this method being invoked.